tests/Unit/Domain/FunctionsOfTime/Test_ReadSpecThirdOrderPiecewisePolynomial.cpp

```diff
@@ -0,0 +1,333 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Framework/TestingFramework.hpp"
+
+#include <array>
+#include <cstddef>
+#include <limits>
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+#include <unordered_map>
+
+#include "DataStructures/DataBox/TagName.hpp"
+#include "Domain/Creators/DomainCreator.hpp"
+#include "Domain/FunctionsOfTime/FunctionOfTime.hpp"
+#include "Domain/FunctionsOfTime/OptionTags.hpp"
+#include "Domain/FunctionsOfTime/PiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/ReadSpecThirdOrderPiecewisePolynomial.hpp"
+#include "Domain/FunctionsOfTime/RegisterDerivedWithCharm.hpp"
+#include "Domain/FunctionsOfTime/Tags.hpp"
+#include "Domain/Structure/ElementId.hpp"
+#include "Framework/TestCreation.hpp"
+#include "IO/H5/AccessType.hpp"
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/File.hpp"
+#include "Informer/InfoFromBuild.hpp"
+#include "Options/Auto.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+#include "Utilities/FileSystem.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace {
+void test_options() noexcept {
+  CHECK(
+      db::tag_name<domain::FunctionsOfTime::OptionTags::FunctionOfTimeFile>() ==
+      "FunctionOfTimeFile");
+  CHECK(db::tag_name<
+            domain::FunctionsOfTime::OptionTags::FunctionOfTimeNameMap>() ==
+        "FunctionOfTimeNameMap");
+
+  const std::string option_string{
+      "CubicFunctionOfTimeOverride:\n"
+      "  FunctionOfTimeFile: TestFile.h5\n"
+      "  FunctionOfTimeNameMap: {Set1: Name1, Set2: Name2}"};
+  using option_tags =
+      tmpl::list<domain::FunctionsOfTime::OptionTags::FunctionOfTimeFile,
+                 domain::FunctionsOfTime::OptionTags::FunctionOfTimeNameMap>;
+  Options::Parser<option_tags> options{""};
+  options.parse(option_string);
+  CHECK((static_cast<std::optional<std::string>>(
```

Comment:
I am confused:  why aren't you just checking `options.get<...FunctionofTimeFile>() == "TestFile.h5"` and `options.get<...FunctionOfTimeNameMap>() == expected_set_names`

Reply:
I agree I can do this more simply as you suggest. I think I needed this extra stuff when my option tags were Auto, or I got compiler errors, but I haven't gone back to be sure that was the error I was getting